app/controllers/miq_ae_tools_controller.rb

```diff
@@ -189,6 +199,49 @@ def review_import
     @message = params[:message]
   end
 
+  def retrieve_git_datastore
+    redirect_options = {:action => :review_git_import}
+    git_url = params[:git_url]
+
+    if git_url.blank?
+      add_flash(_("Please provide a valid git URL"), :error)
+    elsif !MiqRegion.my_region.role_active?("git_owner")
+      add_flash(_("Git Owner role not enabled, enable it in Settings -> Configuration"), :error)
+    else
+      if GitRepository.exists?(:url => git_url)
+        flash_message = <<FLASH
+This repository has been used previously for imports; If you use the same domain it will get deleted and recreated
+FLASH
+        flash_message.chomp!
+        status = :warning
+      else
+        flash_message = "Successfully found git repository, please choose a branch or tag"
+        status = :success
+      end
+
+      git_repo = GitRepository.create(:url => git_url)
+      git_repo.update_authentication(:values => {:userid => params[:git_username], :password => params[:git_password]})
+      git_repo.refresh
```

Comment:
@eclarizio 
In a zone implementation where there are multiple appliances one of them would have the GitOwner role, the refresh would have to be sent to the appliance that has the GitOwner role enabled. Isn't this refreshing the repository on the same appliance. Should this be creating a MiqTask.generic_action_with_callback, with the queue options to get the task to be routed to the correct appliance, with queue_options to be something like
queue_options as = {:class_name => 'GitRepository', :method_name => 'refresh', :instance_id => 6, :role => 'git_owner', :args => [])